### PR TITLE
Add support for vue single file components

### PR DESCRIPTION
**Pull request by Havunen**
I realized vue sfc compiler v3 can also parse vue2 files which is enough to pull out dependencies from them, so I pinned the vue3 compiler as a dependency and that should fix the previous issue. I also merged latest changes to the PR

### CODE REVIEW

**Comment by Havunen** [comment]
This pr relates to: https://github.com/pahen/madge/issues/122  and https://github.com/dependents/node-filing-cabinet/pull/111

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "detective-scss": "^4.0.3",
     "detective-stylus": "^4.0.0",
     "detective-typescript": "^11.1.0",
+    "detective-vue2": ">=1",
```
Comment (XhmikosR) [inline package.json:54]: `>=` -> `^` and 3 digit please

Comment (Havunen) [reply package.json:54]: done

**Comment by Havunen** [comment]
@XhmikosR can we merge these PRs please?

**Comment by Havunen** [comment]
ping @XhmikosR

**Comment by Havunen** [comment]
resolved conflicts again

**Comment by tangtangtangtangtang** [comment]
is there any chance to merge this PR, need vue support too
@Havunen @XhmikosR

**Comment by XhmikosR** [comment]
Sorry, I personally won't merge a PR that adds 45 MB to the deps.

**Comment by Havunen** [comment]
It does not add that much, because it depends on the same deps as defined here to allow referencing them and npm will dedupe tthem.
See https://github.com/dependents/node-precinct/blob/main/package.json#L43-L56
all the code is here: https://github.com/Havunen/detective-vue2/blob/main/index.js
this is the only new dep: https://packagephobia.com/result?p=%40vue%2Fcompiler-sfc

**Comment by Havunen** [comment]
One option is to not add a new package for detective-vue2 and implement this https://github.com/Havunen/detective-vue2/blob/main/index.js code directly here. I tried to follow the same coding pattern as used here

**Comment by XhmikosR** [comment]
Node.js 14 is failing and I see + 25 packages (in this repo so both prod + dev).

Also, even if the size increase is 6.9 MB, it's not negligible, still.

**Comment by XhmikosR** [comment]
@Havunen please cut a new detective-vue2 version with updated deps and rebase this. You are right that most deps should already be in our deps tree, so let's give this a go.

**Comment by Havunen** [comment]
@XhmikosR I rebased the branch and also added 1 more test

**Comment by XhmikosR** [comment]
@Havunen thanks. Can you please add tags to your repo and follow semver in the future?

I also submitted https://github.com/Havunen/detective-vue2/pull/1 which should be a major version bump probably, but feel free to adapt it to your needs.

Then we can land this and I'll cut a new minor release

**Comment by XhmikosR** [comment]
Oh, also feel free to move the repo under the dependents org if you prefer.

**Comment by Havunen** [comment]
I re-applied the change for v2 and invited you to the repository as a collaborator. I cannot move it to dependents organization. I can make you admin there and you can then move it

**Comment by XhmikosR** [comment]
Thanks! I don't see the repo settings tab, unfortunately :/

**Comment by Havunen** [comment]
If you delete your fork I can move the ownership to you abd then you can move it to the org

**Comment by XhmikosR** [comment]
Alright, I deleted my fork. Weird that it's not possible to do it in another way. I also added you to the org, maybe you can move the repo now?

**Comment by Havunen** [comment]
Okay, it seems to work now. I initialized the transfer

**Review by XhmikosR — APPROVED** [review]

**Pull request merged**